test/Dialect/LLHD/Canonicalization/signals.mlir

```diff
@@ -1,4 +1,4 @@
-// RUN: circt-opt %s -cse | FileCheck %s
+// RUN: circt-opt %s -cse -canonicalize | FileCheck %s
```

Comment:
Why was this test running cse before?   Can this be removed to just test one pass in isolation?  Should this file be split in two?

Reply:
We had issues with `llhd.prb` CSE and DCE a while back, because we don't want it to be CSEd as they are not the same in different temporal regions, but only within one. But DCE should still work because canonicalization should also merge blocks which only consist of those dead probe instructions. So we ended up using the `MemRead` trait (which does not do CSE perfectly for this case but good enough). This test should check the probe CSE and DCE behavior.
But I agree that they should be split in two files.